vendor/engines/split_accounts/app/services/split_accounts/order_detail_splitter.rb

```diff
@@ -84,6 +84,7 @@ def build_split_order_detail(split)
       # N+1s because the "fake" order detail will not think it needs to hit the DB.
       copy_associations(split_order_detail)
 
+      split_order_detail.readonly! # Don't accidentally try to write something to the database
```

Comment:
This was not strictly necessary, but seemed like a good safety check for the future.